SU2_CFD/src/output/CFlowOutput.cpp

```diff
@@ -184,16 +184,20 @@ void CFlowOutput::SetAnalyzeSurface(CSolver *solver, CGeometry *geometry, CConfi
 
       for (iVertex = 0; iVertex < geometry->nVertex[iMarker]; iVertex++) {
         iPoint = geometry->vertex[iMarker][iVertex]->GetNode();
-
+        
         if (geometry->node[iPoint]->GetDomain()) {
 
           geometry->vertex[iMarker][iVertex]->GetNormal(Vector);
 
           if (axisymmetric) {
             if (geometry->node[iPoint]->GetCoord(1) != 0.0)
               AxiFactor = 2.0*PI_NUMBER*geometry->node[iPoint]->GetCoord(1);
-            else
-              AxiFactor = 1.0;
+            else{
+              auto neighbor0 = geometry->node[iPoint]->GetPoint(0);
+              auto neighbor1 = geometry->node[iPoint]->GetPoint(1);
+              auto neighbor_normal = geometry->vertex[iMarker][iVertex]->GetNormal_Neighbor();
+              AxiFactor = PI_NUMBER*((neighbor0 != neighbor_normal)? geometry->node[neighbor0]->GetCoord(1) : geometry->node[neighbor1]->GetCoord(1));
+            }
           } else {
```

Comment:
Can you try the following please, I think this problem has a general and robust solution, and indeed significant mass flow may go through this cell (even if the radius is small).
```c++
else {
  AxiFactor = 0.0;
  for (unsigned short iNeigh = 0; iNeigh < geometry->node[iPoint]->GetnPoint(); ++iNeigh) {
    auto jPoint = geometry->node[iPoint]->GetPoint(iNeigh);
    if (geometry->node[jPoint]->GetVertex(iMarker) >= 0) {
      AxiFactor = PI_NUMBER * geometry->node[jPoint]->GetCoord(1);
      break;    
    }
  }
}
```

Reply:
Thanks! That works perfectly. I didn't get it when you suggested it at first